src/PageLayout/PageLayout.tsx

```diff
@@ -55,9 +60,27 @@ const Root: React.FC<React.PropsWithChildren<PageLayoutProps>> = ({
   children,
   sx = {}
 }) => {
+  const {enableStickyPane, disableStickyPane, contentTopRef, contentBottomRef, stickyPaneHeight} = useStickyPaneHeight()
   return (
-    <PageLayoutContext.Provider value={{padding, rowGap, columnGap}}>
-      <Box sx={merge<BetterSystemStyleObject>({padding: SPACING_MAP[padding]}, sx)}>
+    <PageLayoutContext.Provider
+      value={{
+        padding,
+        rowGap,
+        columnGap,
+        enableStickyPane,
+        disableStickyPane,
+        contentTopRef,
+        contentBottomRef
+      }}
+    >
+      <Box
+        style={{
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore TypeScript doesn't know about CSS custom properties
+          '--sticky-pane-height': stickyPaneHeight
+        }}
```

Comment:
@colebemis since this value needs to update in real time while the user is scrolling, I wonder if we can measure how this jsx rendering (where you apply a variable to a `<Box style={{...}}` compares in terms of performance to updating the CSS custom property directly using [setProperty](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty) ([example](https://css-tricks.com/updating-a-css-variable-with-javascript/)).

Reply:
@vdepizzol and I tried this out and it seems like the performance is roughly the same.